### ISSUE TITLE
[toolkit] Error when target folder is not in the project

### ISSUE THREAD

**Issue by Rajith90** [wso2/product-microgateway#714]
### Description:
If we build a project that does not have a target folder, then it gives the error [1]

`org.wso2.apimgt.gateway.cli.exception.CLIInternalException: Error occurred while generating source code for the open API definitions.
	at org.wso2.apimgt.gateway.cli.cmd.BuildCmd.execute(BuildCmd.java:120)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.wso2.apimgt.gateway.cli.cmd.Main.main(Main.java:48)
Caused by: java.nio.file.NoSuchFileException: /var/jenkins_home/workspace/MicroGatewayPipeline/MGW-Projects/bookstore/target/gen
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384)
	at java.nio.file.Files.createDirectory(Files.java:674)
	at org.wso2.apimgt.gateway.cli.utils.GatewayCmdUtils.createGenDirectoryStructure(GatewayCmdUtils.java:906)
	at org.wso2.apimgt.gateway.cli.cmd.BuildCmd.execute(BuildCmd.java:105)
	... 2 more`

### Steps to reproduce:
1. Init a project
2. Add an open API
3. Build the project
4. Commit the project to the github and checkout to different location.
5. Then build the project from new location.

Since when checkout from github the target folder is missing. Hence the issue happens

### Affected Product Version:
3.0.1

### OS, DB, other environment details and versions:
- OS: ANy
- Client:
- DB:
- Env (Docker/K8s): ANy

---

**Comment by praminda** [comment]
I'm working on fixing this as part of the changes I'm doing to make `build` command compatible with new packerina changes

**Comment by praminda** [comment]
Introduced fix will create the target directory on the execution of `build` command. Therefore target directory will no longer be a part of initial mgw project structure.

**Comment by praminda** [comment]
Fixes are added to all version specific branches. Issue will be autoclosed once jballerina branch is merged into master

**Comment by praminda** [comment]
All fixes are merged. Closing the issue to flush the open issue list
